### PR TITLE
Add cloud provider node labels

### PR DESCRIPTION
**Pull request by sttts**
This patch adds a `Labels` function to the cloud provider interface. The kubelet applies the returned labels when creating a new node api object.

**Use Case:**

The cloud environment assigns labels to VMs. These should be automatically available as k8s node labels.

**Background:**

The node labels are matched with the NodeSelector of a pod spec by the scheduler. In addition the kubelet itself will check that its node labels match the NodeSelector before admitting a pod to be launched.

The Mesos scheduler will take the NodeSelector into account when selecting a host for a pod. It uses the Mesos provided slave labels as k8s node labels. The first scheduled pod for a host will trigger the launch of the kubelet. The kubelet creates the api node object. It has to know about the node labels at this point because pods sent from the Mesos executor to the kubelet will be matched against the node labels when a NodeSelector is defined in the pod spec. If labels are not applied yet, such a pod is rejected.

**Alternatives:**
- node-label-controller
  
  Node labels could be added to new nodes asynchronously by some kind of node-label-controller. Unfortunately, being asynchronous means that for a short time the node will exist without the labels. When a pod is scheduled onto that node in this moment, the kubelet will reject it because the NodeSelector does not match.
- kubelet command line flag `--labels`
  
  Node labels could be provided by a command line flag of the kubelet. This way the logic to retrieve the labels from the cloud environments is moved to the launcher of the kubelet. In the case of Mesos this would be the Mesos executor. In the case of other environments it might be some configuration management tool like salt or ansible.
  
  For Mesos this alternative would work as well.

The actual Mesos support to retrieve node labels in its cloud provider will be part of https://github.com/kubernetes/kubernetes/pull/13036.

This seems to have been proposed in #9044

### CODE REVIEW

**Comment by jdef** [comment]
Looks related to #13524

**Comment by ncdc** [comment]
@kubernetes/rh-cluster-infra @kubernetes/rh-platform-management

**Comment by bgrant0607** [comment]
Yes, the approach we were pursuing is described in #13524.

cc @pwittrock @dchen1107

**Comment by jdef** [comment]
TL;DR: The aforementioned PR has suggested using a static pod that runs a forever loop, periodically re-applies node labels via `kubectl --overwrite`.

**Comment by sttts** [comment]
I have implemented initial node creation to the scheduler and node updates in the executor on Registration and Reregistration in #13036. I think this is much simpler than a out-process static pod which must poll the labels from somewhere.

Closing this in favor of #13036.
